tests/job.go

```diff
@@ -129,8 +129,8 @@ func NewHelloWorldJobUDP(host string, port string) *batchv1.Job {
 	// local port is used to catch the reply - any number can be used
 	// we make it different than the port to be safe if both are running on the same machine
 	localPort--
-	check := []string{fmt.Sprintf(`set -x; trap "kill 0" EXIT; x="$(head -n 1 < <(echo | nc -up %d %s %s -i 3 -w 3 & nc -ul %d))"; echo "$x" ; if [ "$x" = "Hello UDP World!" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`,
-		localPort, host, port, localPort)}
+	check := []string{fmt.Sprintf(`set -x; ping -c 1 %s; trap "kill 0" EXIT; x="$(head -n 1 < <(echo | nc -up %d %s %s -i 3 -w 3 & nc -ul %d))"; echo "$x" ; if [ "$x" = "Hello UDP World!" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`,
```

Comment:
General comment - 
1. If the issue is ipv6 one, can we add the ping only to the ipv6 cases?
2. Is there an open issue about the bug? 
IIUC, before using an ipv6 service the users will have to send a request with no port to the service before being able to send a request with a port.
Sounds like a bug that we should document. Does regular pods + k8s services have the same issue as well?

Reply:
> General comment -
> 
> 1. If the issue is ipv6 one, can we add the ping only to the ipv6 cases?

Yes, but that would require me to update these functions with the ip family, as they currently do not care at all about it.

I would rather ping always (simpler code, less changes, it's what we already do for `NewHelloWorldJob`).

> 2. Is there an open issue about the bug?

No. I'll create one.

>    IIUC, before using an ipv6 service the users will have to send a request with no port to the service before being able to send a request with a port.

Could be, I did not search deep enough.

>    Sounds like a bug that we should document. Does regular pods + k8s services have the same issue as well?

What do you mean by regular ? IPv4 ? If so, no. They do not require it.

Reply:
> > General comment -
> > 
> > 1. If the issue is ipv6 one, can we add the ping only to the ipv6 cases?
> 
> Yes, but that would require me to update these functions with the ip family, as they currently do not care at all about it.
> 
> I would rather ping always (simpler code, less changes, it's what we already do for `NewHelloWorldJob`).

I think we should fix `NewHelloWorldJob` as well. It is a bug and it should be clear it is a workaround for ipv6 only.

> 
> > 1. Is there an open issue about the bug?
> 
> No. I'll create one.
> 
> > IIUC, before using an ipv6 service the users will have to send a request with no port to the service before being able to send a request with a port.
> 
> Could be, I did not search deep enough.
> 
> > Sounds like a bug that we should document. Does regular pods + k8s services have the same issue as well?
> 
> What do you mean by regular ? IPv4 ? If so, no. They do not require it.

No, I don't mean ipv4. I mean running the tcp/udp/http server on a pod, not a vm. And creating the NodePort service using the k8s api and not `virtctl expose`.

Reply:
@maiqueb please answer the [comment](https://github.com/kubevirt/kubevirt/pull/4272/commits/370860998bf1853d80fa151b2735b44ad354a38f#r511774774).